lib/charms/prometheus_k8s/v0/prometheus.py

```diff
@@ -375,6 +379,27 @@ def jobs(self):
 
         return scrape_jobs
 
+    def alerts(self):
+        alerts = {}
+        for relation in self._charm.model.relations[self._relation_name]:
+            if len(relation.units) == 0:
+                continue
+
+            alert_rules = json.loads(relation.data[relation.app].get("alert_rules", "{}"))
+            scrape_metadata = json.loads(relation.data[relation.app].get("scrape_metadata", "{}"))
+            if not scrape_metadata:
+                continue
+
+            if alert_rules:
+                alerts[relation.id] = {
+                    "groups": alert_rules["groups"],
```

Comment:
I assume this is the `name` on an item in the `groups` list. Shouldn't it be `group` then?

Reply:
[Alert rules](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/) YAML uses `groups` so I stuck with that.

Reply:
Yeah but it is unclear imo. Like this, you imply an alert can have multiple groups and that ain’t the case, right?

Reply:
I am only trying to imitate the exact same structure of YAML files that Prometheus itself uses, so the code can be easily extended to any future use cases. A Prometheus alerting rules file consists of a list of groups. Each group consists of a list of alerts that are executed sequentially. For this charm library each deployment of the consumer charm produces a single alert rules file on the provider side. This rules file has only one group, but it still needs a `groups` key under whose value is a list of one group. This structure is dictated by Prometheus. At present we only have one group because the convention adopted was to have one rule per file. Hence the consumer charm does not specify any group or its name. It only provides a list of alerts. So all these alerts are grouped into a single group. In the future if it is decided that we want to allow the consumer to provide multiple groups then the current implementation can be easily extended by allow that files with extension `.group` can contain more than one alert, all of which belong to the same group.